pkgs/tools/inputmethods/fcitx5/fcitx5-mozc.nix

```diff
@@ -1,129 +1,145 @@
-{ lib, clangStdenv, fetchFromGitHub, fetchurl, fetchpatch
-, python3Packages, ninja, pkg-config, protobuf, zinnia, qt5, fcitx5
-, jsoncpp, gtest, which, gtk2, unzip, abseil-cpp, breakpad, nixosTests }:
-let
-  inherit (python3Packages) python gyp six;
-  utdic = fetchurl {
-    url = "https://osdn.net/downloads/users/39/39056/mozcdic-ut-20220904.tar.bz2";
-    sha256 = "sha256-pmLBCcw2Zsirzl1PjYkviRIZoyfUz5rpESeABDxuhtU=";
-  };
-  japanese_usage_dictionary = fetchFromGitHub {
-    owner = "hiroyuki-komatsu";
-    repo = "japanese-usage-dictionary";
-    rev = "e5b3425575734c323e1d947009dd74709437b684";
-    sha256 = "0pyrpz9c8nxccwpgyr36w314mi8h132cis8ijvlqmmhqxwsi30hm";
-  };
-  zipcode_rel = "202011";
-  jigyosyo = fetchurl {
-    url = "https://osdn.net/projects/ponsfoot-aur/storage/mozc/jigyosyo-${zipcode_rel}.zip";
-    sha256 = "j7MkNtd4+QTi91EreVig4/OV0o5y1+KIjEJBEmLK/mY=";
-  };
-  x-ken-all = fetchurl {
-    url =
-      "https://osdn.net/projects/ponsfoot-aur/storage/mozc/x-ken-all-${zipcode_rel}.zip";
-    sha256 = "ExS0Cg3rs0I9IOVbZHLt8UEfk8/LmY9oAHPVVlYuTPw=";
-  };
-
-in clangStdenv.mkDerivation {
+{
```

Comment:
It may be nice to move this file to `pkgs/by-name`.

Reply:
I tried to move it to `pkgs/by-name/fc/fcitx5-mozc/package.nix`, but for some reason it broke this line here in `all-packages.nix`:
```nix
fcitx5-mozc-ut = fcitx5-mozc.override { mozc = mozc-ut; };
```

```bash
$ nix build .#fcitx5-mozc-ut
error: undefined variable 'fcitx5-mozc'
```

Is there an additional thing I have to do to make it accessible in `all-packages.nix`? It seems to work just fine for `pkgs/by-name/mo/mozc/package.nix` though.

Reply:
Not sure what is up with the error. Letting the package stay is probably fine. You could also consider moving both. In that case, you'd just put the contents above into `pkgs/by-name/fcitx5-mozc-ut/package.nix` like this:

```nix
{
  fcitx5-mozc,
  mozc-ut,
}:

fcitx5-mozc.override {
  mozc = mozc-ut;
}
```

Reply:
Yeah, it's weird. It's not only affecting `fcitx5-mozc-ut`:

```bash
$ file pkgs/by-name/fc/fcitx5-mozc/package.nix
pkgs/by-name/fc/fcitx5-mozc/package.nix: ASCII text

$ nix build .#fcitx5-mozc
error: flake 'git+file:///tmp/nixpkgs' does not provide attribute 'packages.x86_64-linux.fcitx5-mozc', 'legacyPackages.x86_64-linux.fcitx5-mozc' or 'fcitx5-mozc'

$ file pkgs/by-name/mo/mozc/package.nix
pkgs/by-name/mo/mozc/package.nix: ASCII text

$ nix build .#mozc # <- works
```

I wonder if there's a cache issue here. I also tried `nix build --no-eval-cache .#fcitx5-mozc`, but no dice.

I think I'll keep the legacy path for now, since that's what's recommended anyways:
https://discourse.nixos.org/t/ci-will-soon-enforce-pkgs-by-name-for-new-packages/38098
> There’s no need to migrate existing packages to pkgs/by-name. That will be [done automatically](https://github.com/NixOS/nixpkgs/pull/211832) as the next and final step of RFC 140.

Reply:
@musjj Did you add the file to git? I have forgotten this a couple of times.

Reply:
Argh, thanks for pointing that out 🤦!